api/sendfild/pb/sendfil.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+package api.sendfild.pb;
+option go_package = "github.com/textileio/textile/v2/api/sendfild/pb";
+
+import "google/protobuf/timestamp.proto";
+
+enum MessageState {
+  MESSAGE_STATE_UNSPECIFIED = 0;
+  MESSAGE_STATE_PENDING = 1;
+  MESSAGE_STATE_ACTIVE = 2;
+  MESSAGE_STATE_FAILED = 3;
+}
+
+enum WaitingFilter {
+  WAITING_FILTER_UNSPECIFIED = 0;
+  WAITING_FILTER_WAITING = 1;
+  WAITING_FILTER_NOT_WAITING = 2;
+}
+
+message Txn {
+  string id = 1;
+	string from = 2;
+	string to = 3;
+	int64 amount_nano_fil = 4;
+	string message_cid = 5;
+	MessageState message_state = 6;
+	bool waiting = 7;
+	string failure_msg = 8;
+	google.protobuf.Timestamp created_at = 9;
+	google.protobuf.Timestamp updated_at = 10;
+}
+
+message SendFilRequest {
+  string from = 1;
+  string to = 2;
+  int64 amount_nano_fil = 3;
+  bool wait = 4;
+}
+
+message SendFilResponse {
+  Txn txn = 1;
+}
+
+message TxnRequest {
+  string message_cid = 1;
+  bool wait = 2;
```

Comment:
Both `SendFil` and `Txn` take a `wait` parameter. If `true`, the response won't come until the transaction is active on chain or times out.

Reply:
Yea exactly. We use a reasonable timeout while waiting for the message in the lotus api. If that timeout expires, the error returned here will say that is what happened. The client is free to make another request. The service will also retry any previously timed out waits.